guppylang/prelude/quantum_functional.py

```diff
@@ -0,0 +1,108 @@
+"""Guppy standard module for quantum operations."""
+
+# mypy: disable-error-code="empty-body, misc"
+
+import typing
+
+from guppylang.decorator import guppy
+from guppylang.prelude._internal.compiler.quantum import HSERIES_EXTENSION
+from guppylang.prelude._internal.util import quantum_op
+from guppylang.prelude.angles import angle
+from guppylang.prelude.builtins import owned
+from guppylang.prelude.quantum import quantum, qubit
+
+
+@guppy.hugr_op(quantum, quantum_op("H"))
+def h(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("CZ"))
+def cz(control: qubit @ owned, target: qubit @ owned) -> tuple[qubit, qubit]: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("CX"))
+def cx(control: qubit @ owned, target: qubit @ owned) -> tuple[qubit, qubit]: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("T"))
+def t(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("S"))
+def s(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("X"))
+def x(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("Y"))
+def y(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("Z"))
+def z(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("Tdg"))
+def tdg(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("Sdg"))
+def sdg(q: qubit @ owned) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("ZZMax", ext=HSERIES_EXTENSION))
+def zz_max(q1: qubit @ owned, q2: qubit @ owned) -> tuple[qubit, qubit]: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("Rz"))
+def rz(q: qubit @ owned, angle: angle) -> qubit: ...
+
+
+@guppy.hugr_op(quantum, quantum_op("Rx"))
```

Comment:
N.B. This is broken in main. Both `rz` and `rx` are annoted with `quantum_op("Rz")`